cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/TopicReplicationFactorAnomaly.java

```diff
@@ -90,13 +90,13 @@ public String toString() {
     sb.append("{Detected following topics which have at least one partition with replication factor other than ")
       .append(_targetReplicationFactor)
       .append(" : {fixable : [");
-    if (_badTopicsByFixability.get(true) != null && !_badTopicsByFixability.get(true).isEmpty()) {
-      _badTopicsByFixability.get(true).forEach(t -> sb.append(t).append(", "));
+    if (_topicsWithBadReplicationFactorByFixability.get(true) != null && !_topicsWithBadReplicationFactorByFixability.get(true).isEmpty()) {
+      _topicsWithBadReplicationFactorByFixability.get(true).forEach(t -> sb.append(t).append(", "));
       sb.setLength(sb.length() - 2);
     }
     sb.append("], unfixable : [");
-    if (_badTopicsByFixability.get(false) != null && !_badTopicsByFixability.get(false).isEmpty()) {
-      _badTopicsByFixability.get(false).forEach(t -> sb.append(t).append(", "));
+    if (_topicsWithBadReplicationFactorByFixability.get(false) != null && !_topicsWithBadReplicationFactorByFixability.get(false).isEmpty()) {
```

Comment:
(Applies to similar usage above) Can we avoid this extra checks and the following length-pruning by (1) ensuring that  `_topicsWithBadReplicationFactorByFixability` is created with an empty collection in case no unfixable / fixable topics exist and (2) using `StringJoiner joiner = new StringJoiner(",");` as we do in `GoalViolations#toString()`?